src/main/java/androidx/content/SharedPreferences.kt

```diff
@@ -20,16 +20,27 @@ import android.content.SharedPreferences
 
 /**
  * Allows editing of this preference instance with a call to
- * [SharedPreferences.Editor.apply] to persist the changes.
+ * [SharedPreferences.Editor.apply] or [SharedPreferences.Editor.commit] to persist the changes.
+ * Default behaviour is [SharedPreferences.Editor.apply]
  *
+ * To [SharedPreferences.Editor.apply] changes
  * ```
  * prefs.edit {
  *     putString("key", value)
  * }
  * ```
+ * To [SharedPreferences.Editor.commit] changes
+ * prefs.edit {
+ *     putString("key", value)
+ * }
  */
-inline fun SharedPreferences.edit(action: SharedPreferences.Editor.() -> Unit) {
+inline fun SharedPreferences.edit(
+    commit: Boolean = false,
+    action: SharedPreferences.Editor.() -> Unit) {
     val editor = edit()
     action(editor)
-    editor.apply()
+    when (commit) {
```

Comment:
Use an `if`/`else` here. I don't think the compiler can eliminate `when`s whereas it definitely can for `if`/`else`.